Update DNSSEC07 implementation

tgreenx opened the pull request:
## Purpose

This PR proposes an update of test case DNSSEC07 implementation.

## Context

Test case specification: https://github.com/zonemaster/zonemaster/pull/1425
Test scenarios specification: https://github.com/zonemaster/zonemaster/pull/1432

## Changes

- Update implementation (test case, message tags, profile, DNSSEC module test plan)
- Update unit tests
- Update unit test data

## How to test this PR

Unit tests are updated and should pass.

matsduf reviewed (changes requested):
From the description of zonemaster/zonemaster#1425:

```
After this update, DNSSEC07 should still be run first, and then DNSSEC11. If updated DNSSEC07 outputs DS07_NOT_SIGNED then no other test cases, besides DNSSEC11, in DNSSEC module should be run.
```

I.e. DNSSEC07 and DNSSEC11 should always be run.

Should that be included in the specification? I do not think so, but it should be stated somewhere.

tgreenx commented:
> From the description of [zonemaster/zonemaster#1425](https://github.com/zonemaster/zonemaster/pull/1425):
> 
> ```
> After this update, DNSSEC07 should still be run first, and then DNSSEC11. If updated DNSSEC07 outputs DS07_NOT_SIGNED then no other test cases, besides DNSSEC11, in DNSSEC module should be run.
> ```
> 
> I.e. DNSSEC07 and DNSSEC11 should always be run.
> 
> Should that be included in the specification? I do not think so, but it should be stated somewhere.

For that purpose with have the "Special procedural requirements" section in the specification, which I followed. But it seems it was not entirely updated correctly then.

matsduf commented on lib/Zonemaster/Engine/Test/DNSSEC.pm:
```diff
@@ -1545,10 +1588,6 @@ Readonly my %TAG_DESCRIPTIONS => (
-    NOT_SIGNED => sub {
-        __x    # DNSSEC:NOT_SIGNED
-          'The zone is not signed with DNSSEC.', @_;
-    },
```
The msgid for **NOT_SIGNED** is removed here.

tgreenx commented:
> The **NOT_SIGNED** tag should be cleaned away, but it does no harm. The cleaning could also be done later.

Done (removed). I've also rebased and fixed several conflicts. Please re-review
